xapian-core/cluster/cluster.cc

```diff
@@ -246,12 +236,15 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     for (TermIterator it = doc.termlist_begin(); it != doc.termlist_end(); ++it) {
 	doccount wdf = it.get_wdf();
 	string term = *it;
-	if (wdf < 1 || term[0]=='Z')
+	double termfreq = tlg.get_termfreq(term);
+	/** If term is not a stemmed term or the term indexes only one
+	 *  one document, do not compute TF-IDF scores for them
+	 */
+	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
```

Comment:
I think it'd be better to filter out terms in the `TermListGroup` rather than in `Point`. That way it's easy (ish) for someone to come up with their own filtering, by extending TLG appropriately.